frontend/public/components/operator-hub/operator-hub-subscribe.tsx

```diff
@@ -8,237 +8,216 @@ import { Alert } from 'patternfly-react';
+import { OperatorGroupKind, PackageManifestKind, ClusterServiceVersionLogo, SubscriptionKind, InstallPlanApproval, defaultChannelFor, supportedInstallModesFor } from '../operator-lifecycle-manager';
 import { InstallModeType, isGlobal, installedFor, supports } from '../operator-lifecycle-manager/operator-group';
 import { RadioGroup, RadioInput } from '../radio';
 import { OPERATOR_HUB_CSC_BASE } from '../../const';
 import { getOperatorProviderType } from './operator-hub-utils';
 import { Tooltip } from '../utils/tooltip';
 
-// TODO: Use `redux-form` or React hooks instead of stateful component
-const withFormState = <P extends WithFormStateProps>(Component: React.ComponentType<P>) => {
-  /**
-   * Controlled component which holds form state (https://reactjs.org/docs/forms.html#controlled-components).
-   */
-  return class WithFormState extends React.Component<P, FormData> {
-    static WrappedComponent = Component;
-
-    state = {
+// TODO(alecmerdler): Use `React.useState` hooks instead of stateful component
+export class OperatorHubSubscribeForm extends React.Component<OperatorHubSubscribeFormProps, OperatorHubSubscribeFormState> {
+  constructor(props) {
+    super(props);
+    this.state = {
       targetNamespace: null,
       installMode: null,
       updateChannel: null,
       approval: InstallPlanApproval.Automatic,
     };
+  }
 
-    static getDerivedStateFromProps(nextProps, prevState) {
-      const updateChannel = !_.isEmpty(_.get(nextProps.packageManifest, 'data')) ? defaultChannelFor(nextProps.packageManifest.data) : null;
-      const installMode = prevState.installMode || !_.isEmpty(_.get(nextProps.packageManifest, 'data'))
-        ? installModesFor(nextProps.packageManifest.data)(updateChannel)
-          .filter(({supported}) => supported)
-          .reduce((acc, mode) => mode.type === InstallModeType.InstallModeTypeAllNamespaces
-            ? InstallModeType.InstallModeTypeAllNamespaces
-            : acc,
-          InstallModeType.InstallModeTypeSingleNamespace)
-        : null;
-      let targetNamespace = prevState.targetNamespace || nextProps.targetNamespace;
-      if (installMode === InstallModeType.InstallModeTypeAllNamespaces) {
-        // FIXME(alecmerdler): Will throw error if no global `OperatorGroup`
-        targetNamespace = _.get(nextProps.operatorGroup, 'data', []).find(og => isGlobal(og)).metadata.namespace;
-      }
-
-      return {
-        targetNamespace,
-        installMode: prevState.installMode || installMode,
-        updateChannel: prevState.updateChannel || updateChannel,
-        approval: prevState.approval || InstallPlanApproval.Automatic,
-      };
-    }
-
-    render() {
-      return <Component {...this.props} updateFormState={(state) => this.setState(state)} formState={() => this.state} />;
+  render() {
+    if (!this.props.packageManifest.loaded || _.isEmpty(_.get(this.props.packageManifest, 'data'))) {
```

Comment:
Will this stay loading forever if the package manifest request fails? Would be better to show an error.

OK in as a follow on since it doesn't appear to be a new issue.